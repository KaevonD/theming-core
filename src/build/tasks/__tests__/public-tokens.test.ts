// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { preset, defaultsResolution } from '../../../__fixtures__/common';

import { renderJS, renderSCSS, renderTS } from '../public-tokens';

const propertiesMap = preset.propertiesMap;
const variablesMap = preset.variablesMap;
const publicTokens = preset.exposed;

test('renderJS matches previous snapshot', () => {
  expect(renderJS(defaultsResolution, propertiesMap, publicTokens)).toMatchSnapshot();
});

test('renderSCSS matches previous snapshot', () => {
  expect(renderSCSS(defaultsResolution, variablesMap, propertiesMap, publicTokens)).toMatchSnapshot();
});

test('renderTS matches previous snapshot', () => {
  expect(renderTS(publicTokens)).toMatchSnapshot();
});
