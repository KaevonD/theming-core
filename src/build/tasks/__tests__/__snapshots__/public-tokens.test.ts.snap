// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderJS matches previous snapshot 1`] = `
"export var shadow = \\"var(--shadow-css, grey)\\";
export var buttonShadow = \\"var(--buttonShadow-css, grey)\\";
export var boxShadow = \\"var(--boxShadow-css, grey)\\";
export var lineShadow = \\"var(--lineShadow-css, grey)\\";"
`;

exports[`renderSCSS matches previous snapshot 1`] = `
"$shadow-var: var(--shadow-css, grey);
$buttonShadow-var: var(--buttonShadow-css, grey);
$boxShadow-var: var(--boxShadow-css, grey);
$lineShadow-var: var(--lineShadow-css, grey);"
`;

exports[`renderTS matches previous snapshot 1`] = `
"export const shadow: string;
export const buttonShadow: string;
export const boxShadow: string;
export const lineShadow: string;"
`;
